src/interfaces.ts

```diff
@@ -0,0 +1,55 @@
+export interface LEVLEICredential {
+    i: string;
+    dt: string;
+    LEI: string;
+}
+export interface LE_VLEI_EDGES {
+    d: string,
+    qvi: QVI_VLEI_EDGE
+}
+export interface QVI_VLEI_EDGE {
+    n: string,//issuer of the credential
+    s: "EBfdlu8R27Fbx-ehrqwImnK-8Cm79sqbAQ4MmvEAYqao", //subject of the credential
+}
+
+export interface ECRVLEICredential {
+    i: string;  // issuer
+    dt: string; // date
+    LEI: string; // LEI
+    personLegalName: string; // person's legal name to which the role is assigned
+    engagementContextRole: string; // TODO: find list of roles
```

Comment:
no TODO here, context roles are freeform text

Reply:
I thought I saw somewhere there is an official list of all of the roles that glief describes that are allowed as a role. Could we have a ecr for a company's "technoking"? asking for a [friend](https://www.cnbc.com/2021/12/07/elon-musk-ceo-is-made-up-title-prefers-tesla-technoking.html) lol

Reply:
Lol nice! Someday there will be a technoking ecr 😎

Reply:
the official list will be role codes from ISO.

These are ECRs you can create a `technoking` ECR if you desire.
Head of Standards (ECR)
Managing Director (OOR)

would be two examples of the difference.